src/plugin/places/placesmanager.js

```diff
@@ -162,6 +165,44 @@ plugin.places.PlacesManager.prototype.getPlacesRoot = function() {
 };
 
 
+/**
+ * Iterate over all nodes in the places tree and add all folders to the list.
+ * @param {os.structs.ITreeNode=} opt_node
+ * @private
+ */
+plugin.places.PlacesManager.prototype.addFolders_ = function(opt_node) {
+  var node;
+  if (!opt_node) {
+    node = this.placesRoot_;
+  } else {
+    node = opt_node;
+  }
+  if (node) {
+    if (node.isFolder() && !this.folders_.includes(node)) {
+      this.folders_.push(node);
+    }
+    if (node.getChildren()) {
+      for (var i = 0; i < node.getChildren().length; i++) {
+        var child = node.getChildren()[i];
+        this.addFolders_(child);
+      }
+    }
+  }
+};
```

Comment:
Ignoring the comment about presenting a tree, this is just generically flattening a tree. A generic function like this would be better placed on a static helper package for tree nodes. Also, there are several unnecessary class member items used here. This function could be more simply implemented as `function(ITreeNode, Array<ITreeNode>)` or `function(ITreeNode):Array<ITreeNode>` rather than storing the result of the function as a side effect on a member variable.

However, since we want to present a tree, this is probably no longer necessary.

Reply:
Since I left the flattened tree for now, I did make this change. If there is a better place to put that utility method or you have some feedback on its implementation/use, please let me know.